resources/puppetlabs/lein-ezbake/template/global/tasks/build.rake

```diff
@@ -32,14 +48,82 @@ namespace :pl do
+        else
+          platform_path = "deb/#{platform}/#{repo}"
+        end
+
         FileUtils.mkdir_p("#{pkg_path}/#{platform_path}") unless File.directory?("#{pkg_path}/#{platform_path}")
         # there's no differences in packaging for deb vs ubuntu so picking debian
         # if that changes we'll need to fix that
-        `bash controller.sh debian #{platform}`
+        puts "===================================="
+        puts "Packaging for #{platform}"
+        puts "===================================="
+        stdout, stderr, exitstatus = Pkg::Util::Execution.capture3(%(bash controller.sh debian #{platform}))
+        Pkg::Util::Execution.success?(exitstatus) or raise "Error running packaging: #{stdout}\n#{stderr}"
+        puts "#{stdout}\n#{stderr}"
         FileUtils.cp(Dir.glob("*#{platform}*.deb"), "#{pkg_path}/#{platform_path}")
       end
       FileUtils.cp_r(pkg_path, nested_output)
     end
   end
+  desc "get the property and bundle artifacts ready"
+  task :prep_artifacts, [:output_dir] => "pl:fetch" do |t, args|
+    props = Pkg::Config.config_to_yaml
+    bundle = Pkg::Util::Git.git_bundle('HEAD')
+    FileUtils.cp(props, "#{args[:output_dir]}/BUILD_PROPERTIES")
+    FileUtils.cp(bundle, "#{args[:output_dir]}/PROJECT_BUNDLE")
+  end
+  namespace :jenkins do
+    desc "trigger jenkins packaging job"
+    task :trigger_build, [:auth_string, :job_url] do |t, args|
+      Pkg::Util::RakeUtils.invoke_task("pl:prep_artifacts", "#{Dir.pwd}")
+
+      curl_opts = [
+        '--request POST',
+        "--form file0=@#{Dir.pwd}/BUILD_PROPERTIES",
+        "--form file1=@#{Dir.pwd}/PROJECT_BUNDLE",
+      ]
+      if Pkg::Config.build_pe
+        Pkg::Util.check_var('PE_VER', ENV['PE_VER'])
+        curl_opts << %(--form json='{"parameter": [{"name":"PE_VER", "value":"#{ENV['PE_VER']}"},{"name":"BUILD_PROPERTIES", "file":"file0"},{"name":"PROJECT_BUNDLE", "file":"file1"}]}')
+      else
+        curl_opts << %(--form json='{"parameter": [{"name":"BUILD_PROPERTIES", "file":"file0"},{"name":"PROJECT_BUNDLE", "file":"file1"}]}')
+      end
+      if args[:auth_string] =~ /:/
+        curl_opts << "--user #{args[:auth_string]}"
+        Pkg::Util::Net.curl_form_data("#{args[:job_url]}/build", curl_opts)
+      #assume we're using a token
+      else
+        Pkg::Util::Net.curl_form_data("#{args[:job_url]}/build?token=#{args[:auth_string]}", curl_opts)
+      end
+      Pkg::Util::Net.print_url_info(args[:job_url])
+    end
+    desc "trigger jenkins packaging job with local auth"
+    task :trigger_build_local_auth => "pl:fetch" do
+      if Pkg::Config.build_pe
+        job_url = "https://cinext-jenkinsmaster-enterprise-prod-1.delivery.puppetlabs.net/job/enterprise_various-packaging-jobs_packaging-os-clj_lein-ezbake-generic"
+      else
+        job_url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/job/platform_various-packaging-jobs_packaging-os-clj_lein-ezbake-generic"
+      end
```

Comment:
Is there a way to not hard code this? Or at least DRY it up:
```
if Pkg::Config.build_pe
  jenkins = "cinext-jenkinsmaster-enterprise-prod-1"
  stream = "enterprise"
else
  jenkins = "jenkins-master-prod-1"
  stream = "platform"
end
job_url = "https://#{jenkins}.delivery.puppetlabs.net/job/#{stream}_various-packaging-jobs_packaging-os-clj_lein-ezbake-generic"
```

Reply:
These need to be hardcoded to maintain current functionality for kicking off a build from the command line with an `ezbake build`. I can make the suggested change though.